crates/router/src/core/user.rs

```diff
@@ -2329,14 +2330,15 @@ pub async fn terminate_auth_select(
         .store
         .get_user_authentication_method_by_id(&req.id)
         .await
-        .change_context(UserErrors::InternalServerError)?;
+        .change_context(UserErrors::InvalidUserAuthMethodOperation)
+        .attach_printable("No authentication method found for the id")?;
```

Comment:
`InvalidUserAuthMethodOperation` error should only be thrown when id is not found.
not for every DB error.
use `to_not_found_response`